pkg/api/validation/dynakube/dynakube_name.go

```diff
@@ -34,8 +34,21 @@ func nameViolatesDNS1035(_ context.Context, _ *Validator, dk *dynakube.DynaKube)
 }
 
 func nameTooLong(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
-	dynakubeName := dk.Name
-	if dynakubeName != "" && len(dynakubeName) > dynakube.MaxNameLength {
+	if dk.Name == "" {
+		// Make unit testing easier. This can never happen in an actual cluster.
+		return ""
+	}
+
+	nameLen := len(dk.Name)
+
+	if dk.Extensions().IsDatabasesEnabled() {
+		// Pod names are limited to 63 characters and Kubernetes will cut off characters from the owner resource name to ensure that they can be deployed.
+		// Max length for a Deployment to ensure that nothing gets cut off is: 47 (+9 for pod template hash, +5 for random suffix)
+		// To account for DB SQL executor deployment names using the ID (MaxLength=8), add 1 character to the infix length.
+		nameLen += len(extensions.SQLExecutorInfix) + 1
+	}
+
+	if nameLen > dynakube.MaxNameLength {
```

Comment:
BTW is it unrelated change/bugfix inside this PR?

Reply:
Kind of. I didn't remember the pod name length limitation when I implemented this and thought keeping the deployment name below 64 would be enough. IIRC it was one of the reasons why we shortened from `database-datasource` to `db-datasource`. I could move it to a separate PR, but I think it should be included in the RC.